### PR TITLE
Pass parameters explicitely in QueryPlannerRequest instead of context

### PR DESCRIPTION
**Pull request by Geal**
Follow up to https://github.com/apollographql/router/pull/6108/ to pass query planner arguments directly in the request struct instead of the context

---

**Checklist**

Complete the checklist (and note appropriate exceptions) before the PR is marked ready-for-review.

- [ ] Changes are compatible[^1]
- [ ] Documentation[^2] completed
- [ ] Performance impact assessed and acceptable
- Tests added and passing[^3]
    - [ ] Unit Tests
    - [ ] Integration Tests
    - [ ] Manual Tests

**Exceptions**

*Note any exceptions here*

**Notes**

[^1]: It may be appropriate to bring upcoming changes to the attention of other (impacted) groups. Please endeavour to do this before seeking PR approval. The mechanism for doing this will vary considerably, so use your judgement as to how and when to do this.
[^2]: Configuration is an important part of many changes. Where applicable please try to document configuration examples.
[^3]: Tick whichever testing boxes are applicable. If you are adding Manual Tests, please document the manual testing (extensively) in the Exceptions.

### CODE REVIEW

File: apollo-router/src/query_planner/caching_query_planner.rs [hunk 1]
```diff
@@ -326,20 +322,12 @@ where
                     }
                 };
 
-                let schema = self.schema.api_schema();
-                if let Ok(modified_query) = add_defer_labels(schema, &doc.ast) {
```
Comment (Geal) [inline apollo-router/src/query_planner/caching_query_planner.rs:330]: this is already called inside BridgeQueryPlanner

File: apollo-router/src/query_planner/bridge_query_planner.rs [hunk 2]
```diff
@@ -635,19 +627,9 @@ impl Service<QueryPlannerRequest> for BridgeQueryPlanner {
                         operation_name.as_deref(),
                         Arc::new(QueryHash(hash)),
                     )?;
-                    context
```
Comment (Geal) [inline apollo-router/src/query_planner/bridge_query_planner.rs:638]: I don't think that one was correct (and removing this is probably what triggered the snapshot changes). The defer labels modification was added in https://github.com/apollographql/router/pull/3298 to be able to track which deferred response was linked to which fragment in the query, but this is a modification that should not be apparent anywhere else than in the planner and execution service. If we end up exposing the `ApolloCompiler` instance to user plugins, it should not contain those labels

**Comment by Geal** [comment]
it will conflict with the fix in https://github.com/apollographql/router/pull/6108 because it proposes another way to fix it, but that will be easy to resolve

**Review by goto-bus-stop — APPROVED** [review]
lovely

File: apollo-router/src/query_planner/caching_query_planner.rs (hunk 1 above)
Comment (Geal) [reply apollo-router/src/query_planner/caching_query_planner.rs:344]: we could definitely rely on that. This is used when generating the HTTP response, to know which graphql error has to go in which deferred response (especially when there are multiple deferred fragments at the same level) https://github.com/apollographql/router/blob/ffec3f902bda97912929d8df9e0a447c936ac2ae/apollo-router/src/services/execution/service.rs#L376-L380

Comment (goto-bus-stop) [reply apollo-router/src/query_planner/caching_query_planner.rs:344]: I've filed a followup to do that as ROUTER-850

**Pull request merged**